torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -825,28 +825,31 @@ def lower_binary_eltwise(fn, args):
                 return g.call_function(ttnn.split, args=new_args)
 
             if node.target == torch.ops.aten._to_copy.default:
-                # Keep it if casting to bool type(bool may be problematic)
-                if kwargs["dtype"] in [torch.bool]:
-                    return None
                 # Keep it if the graph output uses this op
                 target_users_ops = [user.target for user in node.users.keys()]
                 if "output" in target_users_ops:
                     return None
-                src_dtype = node.args[0].meta["val"].dtype
-                dst_dtype = kwargs["dtype"]
-                # Some aten op need it to cast specific dtype (ex, index_select)
-                # Keep it if casting from int to float or reverse
+
+                src_dtype, dst_dtype = node.args[0].meta["val"].dtype, kwargs["dtype"]
+
+                # casting to bool type is equivalent to checking if the value is not 0
+                if dst_dtype == torch.bool:
+                    return g.call_function(ttnn.ne, args=(args[0], 0))
```

Comment:
Does the output dtype of `ttnn.ne` bool?

Reply:
I think it should be, but don’t think it is actually

Reply:
The result type of `ttnn.ne` is not bool, it's BF16. But I think it could not need to be.
1. If this bool `to_copy` is used by TTNN op, it all use BF16 computation in TTNN. All we need is to ensure the correct numerical values after casting to bool.
2. If it's return or fallback to torch, the `AddDataMovePass` has handled the correct type after inserting `ttnn.to_torch`
```
%ttnn_to_torch : [num_users=1] = call_function[target=ttnn.decorators.ttnn.to_torch](args = (%ttnn_ne,), kwargs = {dtype: torch.bool})
```